src/snowflake/connector/network.py

```diff
@@ -187,6 +189,54 @@ def is_retryable_http_code(code: int) -> bool:
     )
 
 
+def get_http_retryable_error(status_code: int) -> Error:
+    error_class: Type[Error] = STATUS_TO_EXCEPTION.get(
+        status_code, OtherHTTPRetryableError
+    )
+    return error_class(code=status_code)
+
+
+def raise_okta_unauthorized_error(
+    connection: Optional["SnowflakeConnection"], response: Response
+) -> None:
+    Error.errorhandler_wrapper(
+        connection,
+        None,
+        DatabaseError,
+        {
+            "msg": f"Failed to get authentication by OKTA: {response.status_code}: {response.reason}",
+            "errno": ER_FAILED_TO_CONNECT_TO_DB,
+            "sqlstate": SQLSTATE_CONNECTION_REJECTED,
+        },
+    )
+
+
+def raise_failed_request_error(
+    connection: Optional["SnowflakeConnection"],
+    url: str,
+    method: str,
+    response: Response,
+) -> None:
+    TelemetryService.get_instance().log_http_request_error(
+        "HttpError%s" % str(response.status_code),
+        url,
+        method,
+        SQLSTATE_CONNECTION_WAS_NOT_ESTABLISHED,
+        ER_FAILED_TO_REQUEST,
+        response=response,
+    )
+    Error.errorhandler_wrapper(
+        connection,
+        None,
+        InterfaceError,
+        {
+            "msg": f"{response.status_code} {response.reason}: {method} {url}",
+            "errno": ER_FAILED_TO_REQUEST,
+            "sqlstate": SQLSTATE_CONNECTION_WAS_NOT_ESTABLISHED,
+        },
+    )
+
+
```

Comment:
Pulled out the error raising functions so that they can be re-used by ResultBatch's download.